flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/TableEnvironment.java

```diff
@@ -786,6 +786,24 @@ void createFunction(
      */
     String[] listDatabases();
 
+    /**
+     * Gets the names of all databases registered in the specified catalog.
+     *
+     * @param catalogName specified catalog name
+     * @return A list of the names of all registered databases in the specified catalog.
+     */
+    String[] listDatabases(String catalogName);
+
+    /**
+     * Gets the names of all databases registered in the specified catalog.
+     *
+     * @param catalogName specified catalog name
+     * @param notLike is not like
+     * @param likePattern like pattern
+     * @return A list of the names of all registered databases in the specified catalog.
+     */
+    String[] listDatabases(String catalogName, boolean notLike, String likePattern);
```

Comment:
I don't like the idea of having a boolean flag like this in a public api, in particular this one which is a "level one" top level api. I think we need to reiterate on this. What I propose is:

* Use overloads, for example `listDatabasesLike` and `listDatabasesNotLike`.
* Define a new type to specify the filter, or reuse somehow the `Expression` stack we already use.

Reply:
I prefer use overloads,  `listDatabasesLike` and `listDatabasesNotLike`